common/src/thrift/file_system_master.thrift

```diff
@@ -74,60 +74,53 @@ service FileSystemMasterClientService extends common.TachyonService {
   /**
    * Marks a file as completed.
    */
-  void completeFile( /** the id of the file */ 1: i64 fileId,
+  void completeFile( /** the path of the file */ 1: string path,
       /** the method options */ 2: CompleteFileTOptions options)
     throws (1: exception.TachyonTException e)
 
   /**
-   * Creates a file and returns the id of the file.
+   * Creates a directory
    */
-  i64 create( /** the path of the file */ 1: string path,
-      /** the options for completing the file */ 2: CreateTOptions options)
+  void createDirectory( /** the path of the directory */ 1: string path,
+      /** the method options */ 2: CreateDirectoryTOptions options)
     throws (1: exception.TachyonTException e, 2: exception.ThriftIOException ioe)
 
   /**
-   * Frees the given file from Tachyon and returns whether the free operation succeeded.
+   * Creates a file
    */
-  bool free( /** the id of the file */ 1: i64 fileId,
-      /** whether to free recursively */ 2: bool recursive)
-    throws (1: exception.TachyonTException e)
+  void createFile( /** the path of the file */ 1: string path,
+      /** the options for completing the file */ 2: CreateFileTOptions options)
+    throws (1: exception.TachyonTException e, 2: exception.ThriftIOException ioe)
 
   /**
-   * Returns the file block information for the given file and file block index.
+   * Frees the given file or directory from Tachyon
    */
-  common.FileBlockInfo getFileBlockInfo( /** the id of the file */ 1: i64 fileId,
-      /** the index of the file block */ 2: i32 fileBlockIndex)
+  void free( /** the path of the file or directory */ 1: string path,
+      /** whether to free recursively */ 2: bool recursive)
     throws (1: exception.TachyonTException e)
 
   /**
    * Returns the list of file blocks information for the given file.
    */
-  list<common.FileBlockInfo> getFileBlockInfoList( /** the id of the file */ 1: i64 fileId)
+  list<common.FileBlockInfo> getFileBlockInfoList( /** the path of the file */ 1: string path)
     throws (1: exception.TachyonTException e)
 
   /**
-   * Returns the file id for the given path.
-   */
-  i64 getFileId( /** the path of the file */ 1: string path)
-
-  /**
-   * Returns the file information.
+   * Returns the status of the file or directory
    */
-  FileInfo getFileInfo( /** the id of the file */ 1: i64 fileId)
+  FileInfo getStatus( /** the path of the file or directory */ 1: string path)
     throws (1: exception.TachyonTException e)
 
   /**
-   * If the id points to a file, the method returns a singleton with its file information.
-   * If the id points to a directory, the method returns a list with file information for the
-   * directory contents.
+   * Returns the status of the file or directory, only used internally by servers
    */
-  list<FileInfo> getFileInfoList( /** the id of the file */ 1: i64 fileId)
+  FileInfo getStatusInternal( /** the path of the file or directory */ 1: i64 fileId)
```

Comment:
This still looks like the id, and not the path? Is the plan that this will be removed? be updated to take path? Does there need to be a TODO comment somewhere?